python-mini-projects/projects/convert_Imgs/convertDynamic.py

```diff
@@ -0,0 +1,19 @@
+from PIL import Image
+import sys
+import os
+
+try:
+  im = None
+  for root, dirs, files in os.walk("."):
+    for filename in files:
+        if filename.endswith('.jpg'):
+          im = Image.open(filename).convert("RGB")
+          im.save(filename.replace('jpg', 'png'), "png")
+        elif filename.endswith('.png'):
+          im = Image.open(filename).convert("RGB")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *Image.open*. Currently, there are execution paths that do not contain closure statements, for example, when *convert()* throws an exception. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).